lib/views/chatroom/chatroom_page.dart

```diff
@@ -0,0 +1,170 @@
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:fluttertoast/fluttertoast.dart';
+import 'package:google_fonts/google_fonts.dart';
+import 'package:group_chat_example/constants.dart';
+import 'package:group_chat_example/utils/ui_utils.dart';
+import 'package:group_chat_example/views/chatroom/chatroom_components/chat_bar.dart';
+import 'package:group_chat_example/widgets/spinner.dart';
+import 'package:group_chat_example/widgets/back_button.dart' as BB;
+
+import 'bloc/chatroom_bloc.dart';
+import 'chatroom_components/chat_bubble.dart';
+import 'chatroom_components/chatroom_menu.dart';
+import 'enums/content_enum.dart';
+
+class ChatroomPage extends StatefulWidget {
+  const ChatroomPage({super.key});
+
+  @override
+  State<ChatroomPage> createState() => _ChatroomPageState();
+}
+
+class _ChatroomPageState extends State<ChatroomPage> {
+  ScrollController listScrollController = ScrollController();
```

Comment:
Please mark the variable as final because we are not changing it in the future.

Reply:
Yeah this can be final